ts/sagas/notifications.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * A saga to manage notifications
+ */
+
+import { BasicResponseType } from "italia-ts-commons/lib/requests";
+import { Alert, Platform } from "react-native";
+import { call, Effect, select, takeLatest } from "redux-saga/effects";
+
+import { Installation } from "../../definitions/backend/Installation";
+import { PlatformEnum } from "../../definitions/backend/Platform";
+import { BackendClient } from "../api/backend";
+import { apiUrlPrefix, environment } from "../config";
+import I18n from "../i18n";
+import { SESSION_INITIALIZE_SUCCESS } from "../store/actions/constants";
+import {
+  InstallationState,
+  notificationsInstallationSelector
+} from "../store/reducers/notifications/installation";
+import { sessionTokenSelector } from "../store/reducers/session";
+
+const notificationsPlatform: PlatformEnum = Platform.select({
+  ios: PlatformEnum.apns,
+  android: PlatformEnum.gcm
+});
+
+function notifyNotificationsInstallationError() {
+  // If the application is running in dev mode show an alert
+  if (environment === "DEV") {
+    Alert.alert(
+      I18n.t("notifications.installation.errorTitle"),
+      I18n.t("notifications.installation.errorMessage")
+    );
+  }
+}
+
+/**
+ * This generator function calls the ProxyApi `installation` endpoint
+ */
+function* updateInstallation(): Iterator<Effect> {
+  // Get the notifications installation data from the store
+  const notificationsInstallation: InstallationState = yield select(
+    notificationsInstallationSelector
+  );
+
+  // Get the token from the state
+  const sessionToken: string = yield select(sessionTokenSelector);
+
+  // Check if the we have a session token
+  // TODO: Define what to do when the token is not available or is expired
+  // https://www.pivotaltracker.com/story/show/156839083
+  if (sessionToken) {
+    // Check if the notification server token is available (non available on iOS simulator)
+    if (notificationsInstallation.token) {
+      // Send the request to the backend
+      const backendClient = BackendClient(apiUrlPrefix, sessionToken);
+
+      try {
+        const response:
+          | BasicResponseType<Installation>
```

Comment:
doesn't the type get inferred correctly by TS? btw the type should reflect the return type defined in `backend.ts`

Reply:
Yes i have to replace Installation with NonEmptyString. The problem is the same above, after yield we loose types. I think the problem is going to be solved in the new version of redux-saga that is now at v1.0.0-beta.1.